app/routes/provider.js

```diff
@@ -0,0 +1,35 @@
+import Ember from 'ember';
+import config from 'ember-get-config';
+
+export default Ember.Route.extend({
+    theme: Ember.inject.service(),
+
+    providerIds: config.PREPRINTS.providers.map(provider => provider.id),
+
+    beforeModel(transition) {
+        const {slug} = transition.params.provider;
+
+        if (this.get('providerIds').includes(slug)) {
+            this.set('theme.id', slug);
+        } else {
+            this.set('theme.id', config.PREPRINTS.provider);
+
+            if (slug.length === 5) {
+                this.transitionTo('content', slug);
+            } else {
+                this.replaceWith('page-not-found');
```

Comment:
I am not sure - there is just inconsistent behavior.  A lot of your transitions to `page-not-found` keep the problematic url in the location bar, which I personally like, because I can then look at the url and see what I typed in incorrectly.  If this line is hit, the location bar has the `/preprints/page-not-found` there.